devices/xiaomi.js

```diff
@@ -121,10 +139,93 @@ const tzLocal = {
+            case 'sensor': {
+                const device = Buffer.from(entity.deviceIeeeAddress.substring(2), 'hex');
+                const timestamp = Buffer.allocUnsafe(4);
+                timestamp.writeUint32BE(Date.now()/1000);
+
+                if (value === 'external') {
+                    const params1 = [
+                        ...timestamp,
+                        0x3d, 0x04,
+                        ...device,
+                        ...sensor,
+                        0x00, 0x01, 0x00, 0x55,
+                        0x13, 0x0a, 0x02, 0x00, 0x00, 0x64, 0x04, 0xce, 0xc2, 0xb6, 0xc8,
+                        0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x3d,
+                        0x64,
+                        0x65,
+                    ];
+                    const params2 = [
+                        ...timestamp,
+                        0x3d, 0x05,
+                        ...device,
+                        ...sensor,
+                        0x08, 0x00, 0x07, 0xfd,
+                        0x16, 0x0a, 0x02, 0x0a, 0xc9, 0xe8, 0xb1, 0xb8, 0xd4, 0xda, 0xcf, 0xdf, 0xc0, 0xeb,
+                        0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x3d,
+                        0x04,
+                        0x65,
+                    ];
+
+                    const val1 = [...(aqaraHeader(0x12, params1, 0x02)), ...params1];
+                    const val2 = [...(aqaraHeader(0x13, params2, 0x02)), ...params2];
+
+                    await entity.write('aqaraOpple', {0xfff2: {value: val1, type: 0x41}}, {manufacturerCode: 0x115f});
+                    await entity.write('aqaraOpple', {0xfff2: {value: val2, type: 0x41}}, {manufacturerCode: 0x115f});
+                } else if (value === 'internal') {
+                    const params1 = [
+                        ...timestamp,
+                        0x3d, 0x05,
+                        ...device,
+                        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    ];
+                    const params2 = [
+                        ...timestamp,
+                        0x3d, 0x04,
+                        ...device,
+                        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    ];
+
+                    const val1 = [...(aqaraHeader(0x12, params1, 0x04)), ...params1];
+                    const val2 = [...(aqaraHeader(0x13, params2, 0x04)), ...params2];
+
+                    await entity.write('aqaraOpple', {0xfff2: {value: val1, type: 0x41}}, {manufacturerCode: 0x115f});
+                    await entity.write('aqaraOpple', {0xfff2: {value: val2, type: 0x41}}, {manufacturerCode: 0x115f});
+
+                    await entity.read('hvacThermostat', ['localTemp']);
+                }
+                break;
+            }
+            case 'sensor_temp':
+                if (meta.state['sensor'] === 'external') {
+                    const temperatureBuf = Buffer.allocUnsafe(4);
```

Comment:
Why not use `Buffer.alloc`? (https://www.w3schools.com/nodejs/met_buffer_allocunsafe.asp)

Reply:
the allocUnsafe method works faster than alloc, I'll fix it for alloc